lineage/lineage_graph.py

```diff
@@ -249,7 +273,42 @@ def _update_selected_node_attributes(self, selected_node: str) -> None:
             node.update({'color': self.SELECTED_NODE_COLOR,
                          'title': self.SELECTED_NODE_TITLE + node_title})
 
+    def _get_freshness_and_volume_graph_for_node(self, node: str) -> str:
```

Comment:
I would recommend decoupling this from the main `lineage_graph.py` logic.
It does not really have to know how the graph is being generated, as long as it receives the `html div block with the expected style, title, and img.

Personally, I find matplotlib charts quite outdated, requiring a lot of imperative effort to change and look nice.
I would recommend looking into Plotly or Altair (my favourite), which are also capable of providing interactive viz 👍